### PR TITLE
matt/gen-updates

### PR DESCRIPTION
**Pull request by mattbarker016**
I roughly implemented a smaller **LiveIndicator**, but it will likely need to be tweaked, but the framework to do that is there. I also reworked the **Circle** view, which again needs to be looked at to be right across multiple screens. I think we should get this on master and just work together on the branch to iron out the issues. 

**Update:** Fixed basic errors from merge conflicts. I converted the LiveIndicator UIImageView to the LiveIndicator view. Hopefully color changes will work!

### CODE REVIEW

File: TCAT/AllStopsTableViewController.swift [hunk 1]
```diff
@@ -126,9 +131,9 @@ class AllStopsTableViewController: UITableViewController {
         let section = sectionIndexes[sortedKeys[indexPath.section]]
         let optionsVC = RouteOptionsViewController()
         guard let busStopSelected = section?[indexPath.row]
-            else {
-                print("Could not find bus stop")
-                return
+        else {
```
Comment (myo3) [inline TCAT/AllStopsTableViewController.swift]: prefer `else` on `guard let` line. Just reads better

File: TCAT/Cells/BusStopCell.swift [hunk 2]
```diff
@@ -38,7 +38,7 @@ class BusStopCell: UITableViewCell {
         imageView?.frame = CGRect(x: 10.0, y: 5.0, width: imageWidth, height: imageHeight)
         imageView?.contentMode = .scaleAspectFit
         imageView?.center.y = bounds.height / 2.0
-        imageView?.image = reuseIdentifier == "currentLocation" ? #imageLiteral(resourceName: "location") : #imageLiteral(resourceName: "pin")
+        imageView?.image = reuseIdentifier == Key.Cells.currentLocationIdentifier ? #imageLiteral(resourceName: "location") : #imageLiteral(resourceName: "pin")
```
Comment (myo3) [inline TCAT/Cells/BusStopCell.swift]: i think it's good you're replacing the string with a var  name. however, i think the cell identifier should not be in a global file but rather be a `let` variable at the top of the cell class b/c the only place you're going to use the identifier is in the cell.

File: TCAT/Cells/BusStopTableViewCell.swift [hunk 3]
```diff
@@ -44,7 +44,7 @@ class BusStopTableViewCell: UITableViewCell {
         connectorBottom.backgroundColor = .tcatBlueColor
         contentView.addSubview(connectorBottom)
         
-        statusCircle = Circle(size: .small, color: .tcatBlueColor, style: .outline)
+        statusCircle = Circle(type: .smallOutline, color: .tcatBlueColor)
```
Comment (myo3) [inline TCAT/Cells/BusStopTableViewCell.swift]: why the change? :( i don't see the point of adding this new enum that encapsulates the size and style b/c it makes the Circle class less flexible. Can we go back??

File: TCAT/Cells/RouteTableViewCell.swift [hunk 4]
```diff
@@ -144,33 +143,33 @@ class RouteTableViewCell: UITableViewCell {
...
     func setLiveElements(withFirstDepartDirection firstDepartDirection: Direction?) {
+        
         if let direction = firstDepartDirection {
-            Network.getBusLocations(routeID: String(direction.routeNumber), tripID: direction.tripID, stopID: direction.startLocation.id)
-                .perform(withSuccess: { (result) in
-                    if let busLocation = result.busLocation {
+            
+            Network.getBusLocations([direction]).perform(withSuccess: { (result) in
+                
+                for busLocation in result.busLocations {
```
Comment (myo3) [inline TCAT/Cells/RouteTableViewCell.swift]: why is this a for loop? i only want the live update for the first depart direction so i would think i would only need `result.busLocations.first`. im just confused on why we're going through an array of `busLocation` when I really only want the delay for the bus that corresponds to the first depart direction.

File: TCAT/Cells/SearchResultsCell.swift [hunk 5]
```diff
@@ -39,7 +39,7 @@ class SearchResultsCell: UITableViewCell {
         imageView?.contentMode = .scaleAspectFit
         imageView?.center.y = bounds.height / 2.0
         imageView?.image = #imageLiteral(resourceName: "pin")
-        imageView?.tintColor = reuseIdentifier == "cornellDestinations" ? .tcatBlueColor : .placeColor
+        imageView?.tintColor = reuseIdentifier == Key.Cells.cornellDestinationsIdentifier ? .tcatBlueColor : .mediumGrayColor
```
Comment (myo3) [inline TCAT/Cells/SearchResultsCell.swift]: same comment. think resuseIdentifiers vars should be associated with the cell their in rather than a global Key struct

File: TCAT/Utilities/Network.swift [hunk 6]
```diff
@@ -12,18 +12,24 @@ import CoreLocation
 import GooglePlaces
 
 class Network {
-
-    static let ipAddress = "54.174.47.32" // 10.132.9.134
-
+    
+    /// IP Address of server instance
+    static let ipAddress = "192.168.1.8" // Current Backend IP Address: 54.174.47.32
```
Comment (myo3) [inline TCAT/Utilities/Network.swift]: let's change that back to `54.174.47.32`

File: TCAT/Utilities/Network.swift [hunk 7]
```diff
@@ -81,16 +87,30 @@ class Network {
...
-        request.parameters = [
-            "routeID"   :     routeID,
-            "tripID"    :     tripID,
-            "stopID"    :     stopID
-        ]
-        request.method = .get
+        
+        let departDirections = directions.filter { $0.type == .depart }
+
+        let dictionary = departDirections.map { (direction) -> [String : Any] in
```
Comment (myo3) [inline TCAT/Utilities/Network.swift]: nice use of swifty functional programming

File: TCAT/Utilities/Network.swift [hunk 8]
```diff
@@ -161,15 +181,21 @@ class AllBusStops: JSONDecodable {
...
 
     required init(json: JSON) throws {
+        print("Bus Location JSON:", json)
         if json["success"].boolValue {
-            let data = json["data"]
-            busLocation = parseBusLocation(json: data)
+            if let data = json["data"].array {
+                busLocations = data.map { parseBusLocation(json: $0) }
+            } else {
+                busLocations = []
```
Comment (myo3) [inline TCAT/Utilities/Network.swift]: don't think this is necessary since have default value `[]` for `busLocations`

File: TCAT/Views/CircleLine.swift [hunk 9]
```diff
@@ -42,11 +42,11 @@ class CircleLine: UIView {
     
     // MARK: Position
     
-    private func positionSolidCircle(){
+    private func positionSolidCircle() {
```
Comment (myo3) [inline TCAT/Views/CircleLine.swift]: whoops ^^;

File: TCAT/Views/LiveIndicator.swift [hunk 10]
```diff
@@ -24,39 +29,54 @@ class LiveIndicator: UIView {
     fileprivate let DIM_OPACITY: CGFloat = 0.5
     fileprivate let INTERVAL: TimeInterval = LiveIndicator.INTERVAL
     
+    /// The color to draw the indicator with
+    var color: UIColor = .white
```
Comment (myo3) [inline TCAT/Views/LiveIndicator.swift]: not a huge fan of default values unless they properly describe a default way of initializing (like if you have an `init()`). Think its better if you initialize color before the super.init() calls of both inits, rather than provide a random default value.

File: TCAT/Views/LiveIndicator.swift [hunk 11]
```diff
@@ -100,6 +120,11 @@ class LiveIndicator: UIView {
         }
     }
     
+    func setColor(to color: UIColor) {
+        self.color = color
+        drawViews()
```
Comment (myo3) [inline TCAT/Views/LiveIndicator.swift]: wouldn't this add more layers and views on top of what's there? we should be actually changing the colors of the layer and views, not adding more

File: TCAT/Cells/BusStopCell.swift (hunk 2 above)
Comment (mattbarker016) [reply TCAT/Cells/BusStopCell.swift]: It's used in a few places

File: TCAT/Cells/BusStopTableViewCell.swift (hunk 3 above)
Comment (mattbarker016) [reply TCAT/Cells/BusStopTableViewCell.swift]: I'll revert, but different permutations of stuff are sorta untested

File: TCAT/Cells/RouteTableViewCell.swift (hunk 4 above)
Comment (mattbarker016) [reply TCAT/Cells/RouteTableViewCell.swift]: Resolved with new `/delay` endpoint

File: TCAT/Cells/SearchResultsCell.swift (hunk 5 above)
Comment (mattbarker016) [reply TCAT/Cells/SearchResultsCell.swift]: It's used in multiple files though too

File: TCAT/Views/LiveIndicator.swift (hunk 10 above)
Comment (mattbarker016) [reply TCAT/Views/LiveIndicator.swift]: It's not random, it really is the default for the LiveIndicator. Do you think it should be a default parameter in the init instead?

File: TCAT/Views/LiveIndicator.swift (hunk 11 above)
Comment (mattbarker016) [reply TCAT/Views/LiveIndicator.swift]: Good call

File: TCAT/Views/LiveIndicator.swift (hunk 10 above)
Comment (mattbarker016) [reply TCAT/Views/LiveIndicator.swift]: I removed the default by the variable, left it in parameters
